From d15d4ec4f80a9b761153120798c87f6a07dbd484 Mon Sep 17 00:00:00 2001
From: Liu Yu <liuyu@loongson.cn>
Date: Tue, 6 Sep 2022 11:02:36 +0800
Subject: [PATCH] [loong64][mips64][wasm][simd] Fix SpillAdjacentFpRegisters...

Port commit 8e069d6294d13e3899d55ee7c3c22114ce236008

Change-Id: Ifc21ac1bb2d2b93af07ed7b548204b634f1708b9
Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/3875382
Reviewed-by: Jakob Kummerow <jkummerow@chromium.org>
Commit-Queue: Jakob Kummerow <jkummerow@chromium.org>
Auto-Submit: Liu Yu <liuyu@loongson.cn>
Cr-Commit-Position: refs/heads/main@{#83032}
---
 test/unittests/unittests.status                   | 6 ++++++
 test/unittests/wasm/liftoff-register-unittests.cc | 2 +-
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/test/unittests/unittests.status b/test/unittests/unittests.status
index 2385a0d22d..94761097ba 100644
--- a/test/unittests/unittests.status
+++ b/test/unittests/unittests.status
@@ -108,6 +108,12 @@
   'ParsingTest.TooManyArguments': [SKIP],
 }],  # 'arch == mips'
 
+##############################################################################
+['arch == mips64el', {
+  # TODO(mips-team): mips64 do not allocate odd register in liftoff.
+  'WasmRegisterTest.SpreadSetBitsToAdjacentFpRegs': [SKIP],
+}],  # 'arch == mips64'
+
 ##############################################################################
 ['system == windows and asan', {
   # BUG(893437).
diff --git a/test/unittests/wasm/liftoff-register-unittests.cc b/test/unittests/wasm/liftoff-register-unittests.cc
index f2cd354f26..6a0fd21124 100644
--- a/test/unittests/wasm/liftoff-register-unittests.cc
+++ b/test/unittests/wasm/liftoff-register-unittests.cc
@@ -47,7 +47,7 @@ TEST_F(WasmRegisterTest, SpreadSetBitsToAdjacentFpRegs) {
   // GP reg selection criteria: an even and an odd register belonging to
   // separate adjacent pairs, and contained in kLiftoffAssemblerGpCacheRegs
   // for the given platform.
-#if V8_TARGET_ARCH_S390X || V8_TARGET_ARCH_PPC64
+#if V8_TARGET_ARCH_S390X || V8_TARGET_ARCH_PPC64 || V8_TARGET_ARCH_LOONG64
       LiftoffRegister::from_code(kGpReg, 4),
       LiftoffRegister::from_code(kGpReg, 7),
 #elif V8_TARGET_ARCH_RISCV32 || V8_TARGET_ARCH_RISCV64
-- 
2.35.1

